staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go

```diff
@@ -89,6 +90,35 @@ func Accessor(obj interface{}) (metav1.Object, error) {
 	}
 }
 
+// AsPartialObjectMetadata takes the metav1 interface and returns a partial object.
+// TODO: consider making this solely a conversion action.
+func AsPartialObjectMetadata(m metav1.Object) *metav1alpha1.PartialObjectMetadata {
+	switch t := m.(type) {
+	case *metav1.ObjectMeta:
+		return &metav1alpha1.PartialObjectMetadata{ObjectMeta: *t}
+	default:
+		return &metav1alpha1.PartialObjectMetadata{
+			ObjectMeta: metav1.ObjectMeta{
```

Comment:
This looks prone to slipping.  Seems like separate function and some kind of fuzzer or reflective test is in order.